Sources/UAPusher/Builder/UABuilder.swift

```diff
@@ -0,0 +1,245 @@
+import Vapor
+
+/// Builder class for the request payload
+public final class UABuilder {
+    
+    // MARK: Class fields
+    
+    var audiencePayload: Audience?
+    var deviceTypesPayload: DeviceTypes?
+    var notificationPayload: Notification?
+    var optionsPayload: Options?
+    var inAppPayload: InApp?
+    var messagePayload: Message?
+    var campaignsPayload: Campaigns?
+    
+    // MARK: Initializers
+    
+    /// Init
+    public init() {}
+    
+    // MARK: Audience
+    
+    /// Set audience from a JSON objet
+    ///
+    /// - Parameter audience: JSON
+    /// - Returns: UABuilder
+    public func audience(audience: JSON) -> UABuilder {
```

Comment:
Would it make more sense to have these initializers on the building blocks instead?